activerecord/lib/active_record/railtie.rb

```diff
@@ -128,7 +128,14 @@ class Railtie < Rails::Railtie # :nodoc:
       if config.active_record.delete(:use_schema_cache_dump)
         config.after_initialize do |app|
           ActiveSupport.on_load(:active_record) do
-            filename = File.join(app.config.paths["db"].first, "schema_cache.yml")
+            db_config = ActiveRecord::Base.configurations.configs_for(
+              env_name: ActiveRecord::Base.connection.migration_context.current_environment,
```

Comment:
Since this is a Railtie we can call `Rails.env` instead here. Normally this won't work in ActiveRecord, but since the Railtie is used only by Railties it's safe to call the Rails constant.